src/jit/lsraxarch.cpp

```diff
@@ -1174,6 +1174,55 @@ void Lowering::TreeNodeInfoInitShiftRotate(GenTree* tree)
 }
 
 //------------------------------------------------------------------------
+// TreeNodeInfoInitPutArgReg: Set the NodeInfo for a PUTARG_REG.
+//
+// Arguments:
+//    node                - The PUTARG_REG node.
+//    argReg              - The register in which to pass the argument.
+//    info                - The info for the node's using call.
+//    isVarArgs           - True if the call uses a varargs calling convention.
+//    callHasFloatRegArgs - Set to true if this PUTARG_REG uses an FP register.
+//
+// Return Value:
+//    None.
+//
+void Lowering::TreeNodeInfoInitPutArgReg(
+    GenTreeUnOp* node, regNumber argReg, TreeNodeInfo& info, bool isVarArgs, bool* callHasFloatRegArgs)
+{
+    assert(node != nullptr);
+    assert(node->OperIsPutArgReg());
+    assert(argReg != REG_NA);
+
+    // Each register argument corresponds to one source.
+    info.srcCount++;
+
+    // Set the register requirements for the node.
+    const regMaskTP argMask = genRegMask(argReg);
+    node->gtLsraInfo.setDstCandidates(m_lsra, argMask);
+    node->gtLsraInfo.setSrcCandidates(m_lsra, argMask);
+
+    // To avoid redundant moves, have the argument operand computed in the
+    // register in which the argument is passed to the call.
+    node->gtOp.gtOp1->gtLsraInfo.setSrcCandidates(m_lsra, m_lsra->getUseCandidates(node));
+
+#if FEATURE_VARARG
+    *callHasFloatRegArgs |= varTypeIsFloating(node->TypeGet());
+
+    // In the case of a varargs call, the ABI dictates that if we have floating point args,
+    // we must pass the enregistered arguments in both the integer and floating point registers.
+    // Since the integer register is not associated with this arg node, we will reserve it as
+    // an internal register so that it is not used during the evaluation of the call node
+    // (e.g. for the target).
+    if (isVarArgs && varTypeIsFloating(node))
+    {
+        regNumber targetReg = comp->getCallArgIntRegister(argReg);
+        info.setInternalIntCount(info.internalIntCount + 1);
+        info.addInternalCandidates(m_lsra, genRegMask(targetReg));
+    }
```

Comment:
Personally I don't think that bifurcating the interface is worth the extra `ifdefs`.